include/circt/Dialect/Moore/MooreClasses.td

```diff
@@ -0,0 +1,58 @@
+#ifndef CIRCT_DIALECT_MOORE_MOORECLASSES
+#define CIRCT_DIALECT_MOORE_MOORECLASSES
+
+include "circt/Dialect/Moore/MooreAttributes.td"
+include "circt/Dialect/Moore/MooreDialect.td"
+include "circt/Dialect/Moore/MooreTypes.td"
+include "circt/Dialect/Moore/MooreOps.td"
+include "mlir/IR/OpBase.td"
+include "mlir/IR/SymbolInterfaces.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "mlir/Interfaces/FunctionInterfaces.td"
+include "mlir/Interfaces/CallInterfaces.td"
+include "mlir/Interfaces/ControlFlowInterfaces.td"
+
+def ClassPropertyDeclOp
+    : MooreOp<"class.propertydecl", [Symbol, HasParent<"ClassDeclOp">]> {
+  let summary = "Declare a class property (variable)";
+
+  let arguments = (ins SymbolNameAttr:$sym_name, TypeAttr:$type);
+
+  let results = (outs);
+  // Printed as:  moore.class.propertydecl @foo : !moore.ref<!moore.i32> { ... }
+  let assemblyFormat = [{
+     $sym_name `:` $type  attr-dict
+  }];
+
+  let extraClassDeclaration = [{
+    // Convenience typed accessor for the stored type.
+    ::mlir::Type getPropertyType() { return getTypeAttr().getValue(); }
+  }];
+}
```

Comment:
Since there are probably only a handful of ops for classes, it might make sense to just add them to `MooreOps.td` where all the other ops live. WDYT?

Reply:
There will be a few more (new, delete, call, vcall, property_ref), but in general I don't have a strong opinion. Shall I move this to `MooreOps.td`?